### PR TITLE
VACMS #10010  Rich_Text_Char_Limit_1000 Refactor:

### PR DESCRIPTION
**Pull request by amponce**
## Description

VACMS | PLATFORM | [#10010](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10010)

Description

Component Refactor: Rich Text Char Limit 1000 component to allow for DRUPAL API data to get processed/transformed through data_service. This will allow the component to consume a basic object type.

Acceptance Criteria

- [x] Component renders as expected
- [x]  Component render markup using the transformed data
- [x] data_service will receive JSON:API data and transform it to the format expected by the component
- [x] Existing component tests are revised to test the updated 'pure' component
- [x] The component is added to StoryBook

## Testing done

Unit testing and manual verification

```
 PASS  src/components/rich_text_char_limit_1000/index.test.tsx
  ParagraphRichTextCharLimit1000 with valid data
    ✓ correctly renders ParagraphRichTextCharLimit1000 component (49 ms)
  ParagraphRichTextCharLimit1000 with invalid data
    ✓ does not render ParagraphRichTextCharLimit1000 when data is null (3 ms)

Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        2.246 s
Ran all test suites matching /src\/components\/rich_text_char_limit_1000/i.
✨  Done in 4.69s.
```

## Screenshots

**Component**
Screen Shot 2022-08-02 at 4 36 38 PM

**Storybook**
Screen Shot 2022-08-03 at 12 04 52 PM

## QA steps

What needs to be checked to prove this works?  

**pull branch and load site locally, click on `RichTextCharLimit example` link**

What needs to be checked to prove it didn't break any related things?  
**No merge conflicts and unit tests pass**

### Definition of Done

- [x] Code Quality Tests have passed.
- [x] Acceptance Criteria in related issue are met.
- [x] Added to storybook

### CODE REVIEW

File: src/components/rich_text_char_limit_1000/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+function RichTextCharLimit1000(
+  wysiwygDataService: WysiwygProps,
+  className: string
+): JSX.Element {
+  return (
+    <Wysiwyg
+      id={wysiwygDataService.id}
+      html={wysiwygDataService.field_wysiwyg.processed}
+      className={className}
```
Comment (amponce) [inline src/components/rich_text_char_limit_1000/index.tsx:18]: This is a little strange so I'm going to look for ways to destructure  the data so it's more clear

Comment (amponce) [reply src/components/rich_text_char_limit_1000/index.tsx]: fixed:

```
export const RichTextCharLimit1000 = ({
  id,
  html,
  className,
}: RichTextCharLimit1000Props): JSX.Element => {
  return <Wysiwyg id={id} html={html} className={className} />
}
```

**Comment by amponce** [comment]
I had a bunch of merge conflicts, so hopefully the rebase that I did will resolve everything after merged to main.

File: src/components/wysiwyg/dataService.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import { EntityMetaInfo } from '@/lib/delegators/entityMetaProvider'
 import { ParagraphResourceType, ParagraphWysiwyg } from '@/types/paragraph'
 import { drupalToVaPath, phoneLinks } from '@/utils/helpers'
-import { DetailedHTMLProps, HTMLAttributes } from 'react'
 import Wysiwyg from '@/components/wysiwyg'
 
-const wysiwygDataService = (entity: ParagraphWysiwyg, viewMode: string) => {
+export const wysiwygDataService = (
+  entity: ParagraphWysiwyg,
+  viewMode?: string
```
Comment (penny-lischer) [inline src/components/wysiwyg/dataService.tsx:8]: Do we need viewMode for this component?

**Review by penny-lischer — APPROVED** [review]
LGTM

**Pull request merged**